tmail-backend/jmap/extensions-cassandra/src/main/scala/com/linagora/tmail/james/jmap/firebase/CassandraFirebaseSubscriptionDAO.scala

```diff
@@ -0,0 +1,106 @@
+package com.linagora.tmail.james.jmap.firebase
+
+import java.time.{ZoneOffset, ZonedDateTime}
+
+import com.datastax.oss.driver.api.core.CqlSession
+import com.datastax.oss.driver.api.core.`type`.DataTypes.{TEXT, TIMESTAMP, UUID, frozenSetOf}
+import com.datastax.oss.driver.api.core.cql.{PreparedStatement, Row}
+import com.datastax.oss.driver.api.querybuilder.QueryBuilder.{bindMarker, deleteFrom, insertInto, selectFrom}
+import com.linagora.tmail.james.jmap.model.{DeviceClientId, FirebaseDeviceToken, FirebaseSubscription, FirebaseSubscriptionExpiredTime, FirebaseSubscriptionId}
+import javax.inject.Inject
+import org.apache.james.backends.cassandra.components.CassandraModule
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor
+import org.apache.james.core.Username
+import org.apache.james.jmap.api.change.TypeStateFactory
+import org.apache.james.jmap.api.model.TypeName
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import scala.collection.immutable.Seq
+import scala.jdk.javaapi.CollectionConverters
+
+object CassandraFirebaseSubscriptionTable {
+  val TABLE_NAME = "firebase_subscription"
+  val USER = "user"
+  val DEVICE_CLIENT_ID = "device_client_id"
+  val ID = "id"
+  val EXPIRES = "expires"
+  val TYPES = "types"
+  val TOKEN = "device_token"
```

Comment:
```suggestion
  val TOKEN = "token"
```

Reply:
This `token` is what I used at first, but it does not work.
`Invalid CQL form [token]: needs double quotes...`

I am not sure but looks like `token` is duplicated with other same word in Cassandra queries, change to random `tokenn` works

Reply:
But likely `device_token` is not exact here.
```
It's a token for app (uniq from firebase configs) and device. For example if you create 2 app with two different firebase configs but you request the token from the same device, the service will give you two different tokens.
```
rf: https://stackoverflow.com/questions/60738755/are-fcm-tokens-specific-to-an-app-or-a-device

Maybe I would change to smth like `fcm_token`